xarray/test/test_backends.py

```diff
@@ -1063,3 +1064,42 @@ def test_extract_h5nc_encoding(self):
                           {'least_sigificant_digit': 2})
         with self.assertRaisesRegexp(ValueError, 'unexpected encoding'):
             _extract_nc4_encoding(var, raise_on_invalid=True)
+
+
+@requires_netCDF4
+class TestDataArrayToNetCDF(TestCase):
+
+    def test_dataarray_to_netcdf_no_name(self):
+        original_da = DataArray(np.arange(12).reshape((3, 4)))
+
+        with create_tmp_file() as tmp:
+            original_da.to_netcdf(tmp)
+
+            loaded_da = open_dataarray(tmp)
```

Comment:
Use the context manager to ensure things always get cleaned up (also less code), e.g.,
```
with open_dataaarray(tmp) as loaded_da:
    self.assertDataArrayIdentical(...)
```